espresso/dockerfiles/op-stack/Dockerfile

```diff
@@ -0,0 +1,112 @@
+# OP Stack Dockerfile, simplified from ops/docker/op-stack-go/Dockerfile
+
+# Build arguments
+ARG TARGET_BASE_IMAGE=alpine:3.20
+ARG TARGETOS
+ARG TARGETARCH
+
+# Base builder image
+FROM --platform=$BUILDPLATFORM golang:1.22.7-alpine3.20 AS builder
+
+RUN apk add --no-cache curl tar gzip make gcc musl-dev linux-headers git jq bash
+
+# Install mise for toolchain management
+RUN curl https://mise.run | MISE_INSTALL_PATH=/usr/local/bin/mise sh
+
+# Install yq
+RUN case "$TARGETARCH" in \
+        "amd64") YQ_ARCH="amd64" ;; \
+        "arm64") YQ_ARCH="arm64" ;; \
+        *) YQ_ARCH="amd64" ;; \
+    esac && \
+    wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_$YQ_ARCH
+
+# Install versioned toolchain
+COPY ./mise.toml .
+RUN mise trust && mise install -v -y just && cp $(mise which just) /usr/local/bin/just && just --version
+
+# Copy and download Go dependencies
+COPY ./go.mod /app/go.mod
+COPY ./go.sum /app/go.sum
+WORKDIR /app
+RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build go mod download
+
+# Copy source code
+COPY . /app
+
+# Build arguments for git metadata
+ARG GIT_COMMIT
+ARG GIT_DATE
+
+# Rust builder for Espresso crypto libraries
+FROM --platform=$BUILDPLATFORM rust:1.84.1-alpine3.20 AS rust-builder
+ARG ESPRESSO_NETWORK_GO_VER=0.0.34
```

Comment:
Add a TODO to check the hash of the espresso go library when we switch to the new one. cc @dailinsubjam

Reply:
TODO added: https://github.com/EspressoSystems/optimism-espresso-integration/pull/182/commits/ce1dc2c8f91a7c1400b102fb377e2c592e277b06. Also added an Asana task.